### PR TITLE
initial pr for swipe to cancel prompt

### PR DESCRIPTION
**Pull request by VikashHart**
[Link to Ticket 🎟](https://trello.com/c/BxIRA5T5)

## Description
[//]: # (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)
Created a prompt for the swipe to cancel message

## Type of change

[//]: # (Please delete options that are not relevant.)

- [x] 🌟New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manually tested
- [x] Unit tests added/updated

## Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Update my ticket with relevant information (inProgress or any comments needed)

### CODE REVIEW

**Comment by VikashHart** [comment]
pr not ready yet for merging

File: PhotoBooth/SwipeToCancelPromptPartialModal.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import UIKit
+
+class SwipeToCancelPromptPartialModal: PartialModal {
+    var view: UIView = {
```
Comment (vhart) [inline PhotoBooth/SwipeToCancelPromptPartialModal.swift]: lazy var?

File: PhotoBooth/SwipeToCancelPromptView.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import UIKit
+
+class SwipeToCancelPromptView: UIView {
+
+    private let viewModel: SwipeToCancelPromptViewModel
+
+    lazy var promptLabel: UILabel = {
+        let label = UILabel()
+        label.text = "NewLabel"
```
Comment (vhart) [inline PhotoBooth/SwipeToCancelPromptView.swift]: using view model here right?

**Comment by VikashHart** [comment]
@vhart here is the pr

File: PhotoBooth/SwipeToCancelPromptView.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        label.adjustsFontSizeToFitWidth = true
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    init(viewModel: SwipeToCancelPromptViewModel) {
```
Comment (vhart) [inline PhotoBooth/SwipeToCancelPromptView.swift]: Should be type SwipeToCancelPrompt

File: PhotoBooth/SwipeToCancelPromptView.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import UIKit
+
+class SwipeToCancelPromptView: UIView {
+
+    private let viewModel: SwipeToCancelPromptViewModel
```
Comment (vhart) [inline PhotoBooth/SwipeToCancelPromptView.swift]: Should be the protocol type

**Comment by vhart** [comment]
Compilation failures. You need to build before pushing to verify everything still compiles

**Comment by VikashHart** [comment]
I fixed the build issue

File: PhotoBooth/PhotoShootSetupCardView.swift [hunk 5]
```diff
@@ -2,7 +2,7 @@ import UIKit
 
 class PhotoShootSetupCard: UIView {
 
-    private let viewModel: PhotoShootSetupCardViewModel
+    private let viewModel: SetupPhotoShootViewModeling
```
Comment (vhart) [inline PhotoBooth/PhotoShootSetupCardView.swift]: if the class is PhotoShootSetUpCard, should the view model protocol be `PhotoShootSetUpCardViewModeling`?

**Pull request merged**